ENH: sjoin_nearest (#1865)
geopandas/geopandas: doc/source/docs/user_guide/mergingdata.rst

   cities_with_country.head()

GeoPandas provides two spatial-join functions:

> - ``sjoin()``: joins based on binary predicates (intersects, contains, etc.)
> - ``sjoin_nearest()``: joins based on proximity, with the ability to set a maximum search radius.

Reviewer: ```suggestion
- :func:`sjoin_nearest`: joins based on binary predicates (intersects, contains, etc.)
- :func:`sjoin`: joins based on proximity, with the ability to set a maximum search radius.
```

Binary Predicate Joins
~~~~~~~~~~~~~~~~~~~~~~

> Binary predicte joins are available via ``sjoin()``.

Reviewer: ```suggestion
Binary predicate joins are available via :func:`sjoin`.
```

Note more complicated spatial relationships can be studied by combining geometric operations with spatial join.
To find all polygons within a given distance of a point, for example, one can first use the :meth:`~geopandas.GeoSeries.buffer` method to expand each
point into a circle of appropriate radius, then intersect those buffered circles with the polygons in question.

Nearest Joins
~~~~~~~~~~~~~

> Proximity based joins can be done via ``sjoin_nearest()``.

Reviewer: ```suggestion
Proximity-based joins can be done via :func:`sjoin_nearest`.
```

> ``sjoin_nearest()`` shares the ``how`` argument with ``sjoin()`` but does not support the ``"inner"`` option.

Reviewer: ```suggestion
:func:`sjoin_nearest` shares the ``how`` argument with :func:`sjoin` but does not support the ``"inner"`` option.
```

> ``sjoin_nearest()`` shares the ``how`` argument with ``sjoin()``, however it does not support an ``op`` argument since no binary predicates are applied.

Reviewer 2: ```suggestion
``sjoin_nearest()`` shares the ``how`` argument with ``sjoin()``.
```

I don't think you need to note `op` / `predicates` here since there isn't an expectation that they'd be supported.

> ``sjoin_nearest()`` includes two additional arguments: ``max_distance`` and ``distance_col``.

Reviewer: ```suggestion
:func:`sjoin_nearest` includes two additional arguments: ``max_distance`` and ``distance_col``.
```

> ``sjoin_nearest()`` does introduce two additional arguments: ``max_distance`` and ``distance_col``.

Reviewer 2: ```suggestion
``sjoin_nearest()`` includes two additional arguments: ``max_distance`` and ``distance_col``.
```

**max_distance**

The ``max_distance`` argument specifies a maximum search radius for matching geometries. This can have a considerable performance impact in some cases.
If you can, it is highly recommended that you use this parameter.

**distance_col**

> If set, a column containing the computed distances for matching geometries will be inserted into the resultant joined GeoDataFrame under the given name.

Reviewer 2: ```suggestion
If set, the resultant GeoDataFrame will include a column with this name containing the computed distances between an input geometry and the nearest geometry.
```
